json_rpc/private/jrpc_sys.nim

```diff
@@ -142,6 +143,11 @@ type
 createJsonFlavor JrpcSys,
   requireAllFields = false
 
+ResponseError.useDefaultSerializationIn JrpcSys
```

Comment:
or `JrpcSys.useDefaultSerializationFor(ResponseError, RequestTx, ...)`, right?

My preferred option would actually be `type RequestTx {.json: [JrpcSys].}` or something along those lines that but I suspect we might run into macro constraints implementing that..

Reply:
> JrpcSys.useDefaultSerializationFor(ResponseError, RequestTx, ...), right?

No, because only `ResponseError` using both reader dan writer, while the rest use only reader or writer, not both.

Reply:
> My preferred option would actually be type RequestTx {.json: [JrpcSys].} or something along those lines that but I suspect we might run into macro constraints implementing that..

I think it's doable. We can use simple template custom pragma instead of macro. The macro part already in nim-serialization. So it's still fall within nim-serialization capabilities.